src/super_gradients/training/sg_model/sg_model.py

```diff
@@ -911,7 +917,7 @@ def train(self, training_params: dict = dict()):  # noqa: C901
 
                 # IN DDP- SET_EPOCH WILL CAUSE EVERY PROCESS TO BE EXPOSED TO THE ENTIRE DATASET BY SHUFFLING WITH A
                 # DIFFERENT SEED EACH EPOCH START
-                if self.multi_gpu == MultiGPUMode.DISTRIBUTED_DATA_PARALLEL:
+                if self.multi_gpu == MultiGPUMode.DISTRIBUTED_DATA_PARALLEL and hasattr(self.train_loader, "sampler") and hasattr(self.train_loader.sampler, "set_epoch"):
```

Comment:
i don't like this logic. this sub-sub-parameter is overriding the multi GPU mode?

Reply:
This is because for distributed samplers we had to call set_epoch each time we start a new epoch (so it would re-shuffle the data).
But sometimes we don't implement set epoch, or even use sampler (i.e use batch sampler), so we need to check before calling set_epoch.